linalg/benches/x86_64.rs

```diff
@@ -0,0 +1,133 @@
+#![allow(dead_code, non_upper_case_globals, unused_macros, non_snake_case, unused_assignments)]
+
+use std::arch::asm;
+
+mod nano;
+
+#[repr(C, align(64))]
+struct Floats([f32; 4096]);
+const _F32: Floats = Floats([12.; 4096]);
+const F32: *const f32 = (&_F32) as *const Floats as *const f32;
+
+lazy_static::lazy_static! {
+    static ref TICK: f64 = unsafe { b8192!(asm!("or rax, rax", out("rax") _)) };
+}
+
+macro_rules! kloop {
+    ($filter: expr, $geo: literal, $n: expr, $path: literal, $ww: expr) => {
+        let label = $path.split("/").last().unwrap().split_once(".").unwrap().0;
+        let full_label = format!("{:8} {:40}", $geo, label);
+        if full_label.contains($filter.unwrap_or("")) {
+            let time = b2!({
+                let mut p = F32;
+                let mut q = F32;
+                r128!(asm!(include_str!(concat!("../x86_64/fma/", $path)),
+						 inout("rax") p, inout("rcx") q,
+				out("zmm0") _, out("zmm1") _, out("zmm2") _, out("zmm3") _,
+                out("zmm4") _, out("zmm5") _, out("zmm6") _, out("zmm7") _,
+                out("zmm8") _, out("zmm9") _, out("zmm10") _, out("zmm11") _,
+                out("zmm12") _, out("zmm13") _, out("zmm14") _, out("zmm15") _,
+                out("zmm20") _, out("zmm21") _, out("zmm22") _, out("zmm23") _,
+                out("zmm24") _, out("zmm25") _, out("zmm26") _, out("zmm27") _,
+                ));
+            }) / 128.;
+
+
+            println!("{} {:3.0}% ({:>5.2 }/{:3 } cy) {:.2} GFLOP/s", full_label, ($n as f64 / $ww as f64) / time * 100. * *TICK, time / *TICK, $n as f64 / $ww as f64, $n as f64 / time / 1e9 );
+        }
+    }
+}
+
+unsafe fn packed_packed_2x6(f: Option<&str>) {
```

Comment:
I would prefer we keep the size part of the kernel name in terms of item count and not register, so please call these 16x6 instead of 2x6, etc. These ids are popping here and there upstream (in the command line dumps for instance) where it generates a lot of cognitive cost to remember the register size on multiple architectures...

Reply:
Did you plan on using the same top-level template for avx2/2x5 and avx512/2x5 ? I understand it would make sense in that situation to switch the naming... If we want to go this way, we will need to put an indirection/renaming at some point, I really don't want to have to remember register sizes when looking at a high level profile of a model.

Reply:
Yeah, I've gone back and forth a few times. It makes more sense from a kernel design to go with register-based sizes, but I understand that it's a higher cognitive load. I.e., when comparing AVX-512 to AVX-2 the proper comparison is 32x6 vs 16x6 to keep "layouts" same, not just using larger registers. Otherwise the load/compute ratio gets skewed.